mvpinternshipapp.client/src/components/DataTable.jsx

```diff
@@ -0,0 +1,55 @@
+import { Button, Loader } from "semantic-ui-react";
+import PropTypes from 'prop-types';
+
+export default function DataTable({
+    headers,
+    dataCellsMapper,
+    data,
+    loading,
+    emptyMessage,
+    onClickEdit,
+    onClickDelete,
+}) {
+    const createRow = (dataObj) => {
+        return (
+            <tr key={dataObj.id} >
+                {dataCellsMapper(dataObj)}
+                <td>
+                    <Button color='yellow' onClick={() => onClickEdit(dataObj)}>Edit</Button>
+
+                    <Button color='red' onClick={() => onClickDelete(dataObj)}>Delete</Button>
+                </td>
+
+            </tr>
+        )
+    }
+    if (loading) {
```

Comment:
Is there a reason why you didn't use any hooks in this one? For example, you can use **useEffect** first and then handle the return properly. We can discuss this on Monday so I can explain it.